File: 1116-maximum-level-sum-of-a-binary-tree/1116-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        vector <TreeNode* > Queue;
        Queue.push_back(root);
        vector <TreeNode* >& queue = Queue;
        int maxLevel = 1;
        int level = 1;
        int maxLevelSum = root->val;
        while(!Queue.empty())
        {
            int levelSum = 0;
            vector <TreeNode* > nextQueue;
            for(TreeNode* node: Queue)
            {
                if(node->left)
                    nextQueue.push_back(node->left);
                if(node->right)
                    nextQueue.push_back(node->right);
                levelSum += node->val;
            }
            if(levelSum > maxLevelSum)
            {
                maxLevelSum = levelSum;
                maxLevel = level;
            }
            level++;
            queue = nextQueue;
        }
        return maxLevel;
    }
};

File: 1116-MaximumLevelSumOfABinaryTree/1116-MaximumLevelSumOfABinaryTree.py
# Last updated: 7/24/2025, 8:28:11 AM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        queue = [root]
        maxLevelSum = root.val
        maxLevel = 1
        level = 1
        while queue:
            nextQueue = []
            levelSum = 0
            
            for node in queue:
                if node.left:
                    nextQueue += [node.left]
                if node.right:
                    nextQueue += [node.right]
                levelSum += node.val

            if levelSum > maxLevelSum:
                maxLevelSum = levelSum
                maxLevel = level
            queue = nextQueue
            level += 1

        
        return maxLevel
